packages/admin-next/dashboard/src/routes/discounts/create/components/create-discount-form/create-discount-details.tsx

```diff
@@ -11,56 +27,642 @@ type CreateDiscountPropsProps = {
+                    <Form.Control>
+                      <Textarea {...field} />
+                    </Form.Control>
+                    <Form.ErrorMessage />
+                  </Form.Item>
+                )
+              }}
+            />
+
+            <Form.Field
+              control={form.control}
+              name="is_dynamic"
+              render={({ field: { value, onChange, ...field } }) => {
+                return (
+                  <Form.Item>
+                    <div className="flex gap-2">
+                      <Form.Control>
+                        <Checkbox
+                          checked={value}
+                          onCheckedChange={(s) => onChange(s === true)}
+                          {...field}
+                        />
+                      </Form.Control>
+                      <Form.Label tooltip={t("discounts.templateHint")}>
+                        {t("discounts.isTemplateDiscount")}
+                      </Form.Label>
+                    </div>
+                    {/*<Form.ErrorMessage />*/}
+                  </Form.Item>
+                )
+              }}
+            />
+          </div>
+
+          {/* CONFIGURATIONS */}
+          <div className="flex flex-col gap-y-8">
+            <div className="flex flex-col gap-y-1">
+              <Text
+                size="large"
+                leading="compact"
+                className="text-ui-fg-base"
+                weight="plus"
+              >
+                {t("fields.configurations")}
+              </Text>
+              <Text
+                size="small"
+                leading="compact"
+                className="text-ui-fg-subtle"
+              >
+                <Trans
+                  t={t}
+                  i18nKey="discounts.codeHint"
+                  components={[<br key="break" />]}
+                />
+              </Text>
+            </div>
+
+            <div className="flex flex-col gap-y-4">
+              <Form.Field
+                control={form.control}
+                name="start_date_enabled"
+                render={({ field: { value, onChange, ...field } }) => (
+                  <Form.Item>
+                    <div className="flex items-center justify-between">
+                      <Form.Label tooltip="todo">
+                        {t("discounts.hasStartDate")}
+                      </Form.Label>
+                      <Form.Control>
+                        <Switch
+                          {...field}
+                          checked={!!value}
+                          onCheckedChange={onChange}
+                        />
+                      </Form.Control>
+                    </div>
+                    <Form.Hint className="!mt-1">
+                      {t("discounts.startDateHint")}
+                    </Form.Hint>
+                    <Form.ErrorMessage />
+                  </Form.Item>
+                )}
+              />
```

Comment:
This applies to all "sections" with a Switch. 

The intended design is that you click the switch to show the input. If you then disable the Switch it should hide the input and then the value should be undefined when submitting.

Then in the Edit form if the value exists on the Discount then it should be checked per default.

Reply:
Should be good now